### PR TITLE
Changed: If fetch hints were not provided throw an exception instead of returning null

### PR DESCRIPTION
**Pull request by joeferner**

### CODE REVIEW

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java [hunk 1]
```diff
@@ -186,17 +184,13 @@ public static Vertex createFromIteratorValue(
 
     @Override
     public Iterable<Edge> getEdges(Direction direction, String label, FetchHints fetchHints, Authorizations authorizations) {
-        if (!getFetchHints().validateHasEdgeFetchHints(direction)) {
-            return null;
-        }
+        getFetchHints().validateHasEdgeFetchHints(direction);
```
Comment (mwizeman) [inline accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java]: Seems like we need a new method `FetchHints.validateHasEdgeFetchHints(Direction, Label...)`

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java [hunk 2]
```diff
@@ -186,17 +184,13 @@ public static Vertex createFromIteratorValue(
...
         return getGraph().getEdges(getEdgeIds(direction, labelToArrayOrNull(label), authorizations), fetchHints, authorizations);
     }
 
     @Override
     public Iterable<String> getEdgeIds(Direction direction, String label, Authorizations authorizations) {
-        if (!getFetchHints().validateHasEdgeFetchHints(direction)) {
-            return null;
-        }
+        getFetchHints().validateHasEdgeFetchHints(direction);
```
Comment (mwizeman) [inline accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java]: See above about checking the label

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java [hunk 3]
```diff
@@ -207,17 +201,13 @@ public static Vertex createFromIteratorValue(
 
     @Override
     public Iterable<Edge> getEdges(Direction direction, final String[] labels, FetchHints fetchHints, Authorizations authorizations) {
-        if (!getFetchHints().validateHasEdgeFetchHints(direction)) {
-            return null;
-        }
+        getFetchHints().validateHasEdgeFetchHints(direction);
```
Comment (mwizeman) [inline accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java]: See above about checking the labels

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java [hunk 4]
```diff
@@ -207,17 +201,13 @@ public static Vertex createFromIteratorValue(
...
         return getGraph().getEdges(getEdgeIdsWithOtherVertexId(null, direction, labels, authorizations), fetchHints, authorizations);
     }
 
     @Override
     public Iterable<String> getEdgeIds(final Direction direction, final String[] labels, final Authorizations authorizations) {
-        if (!getFetchHints().validateHasEdgeFetchHints(direction)) {
-            return null;
-        }
+        getFetchHints().validateHasEdgeFetchHints(direction);
```
Comment (mwizeman) [inline accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java]: See above about checking the labels

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java [hunk 5]
```diff
@@ -249,17 +235,13 @@ public static Vertex createFromIteratorValue(
 
     @Override
     public Iterable<Edge> getEdges(final Vertex otherVertex, Direction direction, String label, FetchHints fetchHints, Authorizations authorizations) {
-        if (!getFetchHints().validateHasEdgeFetchHints(direction)) {
-            return null;
-        }
+        getFetchHints().validateHasEdgeFetchHints(direction);
```
Comment (mwizeman) [inline accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java]: See above about checking the label

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java [hunk 6]
```diff
@@ -249,17 +235,13 @@ public static Vertex createFromIteratorValue(
...
         return getGraph().getEdges(getEdgeIdsWithOtherVertexId(otherVertex.getId(), direction, labelToArrayOrNull(label), authorizations), fetchHints, authorizations);
     }
 
     @Override
     public Iterable<String> getEdgeIds(Vertex otherVertex, Direction direction, String label, Authorizations authorizations) {
-        if (!getFetchHints().validateHasEdgeFetchHints(direction)) {
-            return null;
-        }
+        getFetchHints().validateHasEdgeFetchHints(direction);
```
Comment (mwizeman) [inline accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java]: See above about checking the label

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java [hunk 7]
```diff
@@ -270,17 +252,13 @@ public static Vertex createFromIteratorValue(
 
     @Override
     public Iterable<Edge> getEdges(final Vertex otherVertex, Direction direction, String[] labels, FetchHints fetchHints, Authorizations authorizations) {
-        if (!getFetchHints().validateHasEdgeFetchHints(direction)) {
-            return null;
-        }
+        getFetchHints().validateHasEdgeFetchHints(direction);
```
Comment (mwizeman) [inline accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java]: See above about checking the labels

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java [hunk 8]
```diff
@@ -270,17 +252,13 @@ public static Vertex createFromIteratorValue(
...
         return getGraph().getEdges(getEdgeIdsWithOtherVertexId(otherVertex.getId(), direction, labels, authorizations), fetchHints, authorizations);
     }
 
     @Override
     public Iterable<String> getEdgeIds(final Vertex otherVertex, final Direction direction, final String[] labels, final Authorizations authorizations) {
-        if (!getFetchHints().validateHasEdgeFetchHints(direction)) {
-            return null;
-        }
+        getFetchHints().validateHasEdgeFetchHints(direction);
```
Comment (mwizeman) [inline accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java]: See above about checking the labels

File: inmemory/src/main/java/org/vertexium/inmemory/InMemoryVertex.java [hunk 9]
```diff
@@ -54,8 +50,7 @@ public InMemoryVertex(
     @Override
     public Iterable<EdgeInfo> getEdgeInfos(Direction direction, final String[] labels, Authorizations authorizations) {
         if (!getFetchHints().isIncludeEdgeRefs()) {
```
Comment (mwizeman) [inline inmemory/src/main/java/org/vertexium/inmemory/InMemoryVertex.java]: Should this be using `getFetchHints().validateHasEdgeFetchHints(direction)` instead? If so, see above about checking the labels too.

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java (hunk 1 above)
Comment (joeferner) [reply accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java]: done

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java (hunk 2 above)
Comment (joeferner) [reply accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java]: done

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java (hunk 3 above)
Comment (joeferner) [reply accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java]: done

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java (hunk 4 above)
Comment (joeferner) [reply accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java]: done

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java (hunk 5 above)
Comment (joeferner) [reply accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java]: done

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java (hunk 6 above)
Comment (joeferner) [reply accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java]: done

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java (hunk 7 above)
Comment (joeferner) [reply accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java]: done

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java (hunk 8 above)
Comment (joeferner) [reply accumulo/src/main/java/org/vertexium/accumulo/AccumuloVertex.java]: done

File: inmemory/src/main/java/org/vertexium/inmemory/InMemoryVertex.java (hunk 9 above)
Comment (joeferner) [reply inmemory/src/main/java/org/vertexium/inmemory/InMemoryVertex.java]: done

**Comment by joeferner** [comment]
@mwizeman done

**Pull request merged**